### PR TITLE
force finallap to use 32x32 sprite size for title screen

### PR DESCRIPTION
**Pull request by DavidHaywood**
this is an odd one, other system2 games clearly show that you can select between 16x16 and 32x32 mode with a bit in each sprite definition in the spriteram.

finallap needs 32x32 mode for the title screen, but fails to set the bit on any of the sprites. 

I don't think finallap ever needs 16x16 sprites (although more testing is needed)

finallap3 correctly sets it, and it is needed for cars ingame there

### CODE REVIEW

File: src/mame/video/namcos2_sprite.cpp [hunk 1]
```diff
@@ -23,10 +23,11 @@
 
 DEFINE_DEVICE_TYPE(NAMCOS2_SPRITE, namcos2_sprite_device, "namcos2_sprite", "Namco System 2 Sprites (C106,C134,C135,C146)")
 
-namcos2_sprite_device::namcos2_sprite_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock) :
+namcos2_sprite_device::namcos2_sprite_device(const machine_config& mconfig, const char* tag, device_t* owner, u32 clock) :
```
Comment (cuavas) [inline src/mame/video/namcos2_sprite.cpp:26]: Please don’t go changing the spacing like this – there’s a reason for putting right-associative things to the right of the space.

Comment (DavidHaywood) [reply src/mame/video/namcos2_sprite.cpp:26]: visual studio defaults to this format, MAME seems to be at odds with most other things here (and annoyingly VS seems to like resetting this to defaults from time to time, I'll change it.. again)

**Comment by happppp** [comment]
FYI, finallap rear view mirror has glitchy sprites if 32x32 is not enforced.

**Comment by DavidHaywood** [comment]
so you're saying this improves the real mirror too?

**Comment by Gurudumps** [comment]
Since it was the first game on System 2, Final Lap uses the very early version PCB. All later games use a slightly different board. So not everything on later games can be applied to the first rev of the hardware and vise-versa. Namco obviously realised the tile size needed to be switchable so they updated the hardware to support it.

**Comment by DavidHaywood** [comment]
yeah, that crossed my mind too.  Kale says it is the same 4 chips listed in the file doing the sprites, even on Final Lap 1, but maybe some of that tile splitting was in external logic.

**Comment by angelosa** [comment]
https://www.arcade-museum.com/manuals-videogames/F/FinalLap.pdf

Pages 76-78
There's a good chunk of flip flops in here, which I'm quite positive they are rearranged on System 2 and Metal Hawk.

To my understanding LINE* is zoomy while SIZE* is zoomx, page 77 specifically shows a SIZE2 so the 16x16 hookup is present. 
I'm still not great at routing schems to actual bit positions but it looks to me that there's no DATA13 / LINE5 / SIZE5 as suggested by the driver notes, therefore SIZE2 should be `word#1 D13` here?

**Comment by Gurudumps** [comment]
Four Trax uses the same early rev board (I documented it in the src), which actually has no graphical issues in MAME. Maybe there are some Four Trax specific fixes in MAME that can be applied to Final Lap as well.

**Comment by angelosa** [comment]
> Four Trax uses the same early rev board (I documented it in the src), which actually has no graphical issues in MAME. Maybe there are some Four Trax specific fixes in MAME that can be applied to Final Lap as well.

It just reuses the same machine config, so this fix applies there as well.

**Comment by Gurudumps** [comment]
Is there any hope of a fix for the sound in-game on Suzuka 8 Hours 2? That's been broken like forever. 
On my working PCB if both voice ROMs are removed from the PCB the in-game sound is almost exactly like in MAME, which suggests the voice ROMs are incorrectly mapped or a mirror mapping is missing.
It's exactly the same PCB as Lucky and Wild.

**Comment by DavidHaywood** [comment]
I think it's something in the C140 emulation itself, ROM accesses look normal, and it is getting data from the ROM.  Sound isn't something I'm great with at the best of times though.

My more immediate priority is looking at the Metal Hawk ROZ layer and fixing up the wraparound disable support as some levels really aren't playable as things stand.  Beyond that, Final Lap R / Speed Racer use the same ROZ chip for the road, but in line mode, where again the emulation doesn't currently hold up at all.

Fourtrax, best I can see, never tries to use 16x16 sprites, so works the same before/after the change to force 32x32. I wonder if it was just a design / production flaw in that early board causing them to never work even if they should have done.

**Pull request merged**